networking/benches/stream_reader.rs

```diff
@@ -0,0 +1,219 @@
+        use std::net::TcpListener;
+
+        debug!(sender_log, "Starting listening");
+        let listener = TcpListener::bind("127.0.0.1:0").unwrap();
+        let local_addr = listener.local_addr().unwrap();
+        addr_send.send(local_addr).unwrap();
+        let log = sender_log.new(o!("sender" => local_addr));
+        debug!(log, "Listening");
+
+        listener.set_nonblocking(true).unwrap();
+
+        let precompute_key = crypto_remote.precompute_key;
+        let nonce = crypto_remote.nonce_pair.local;
+
+        loop {
+            match listener.accept() {
+                Ok((socket, _)) => {
+                    debug!(log, "connected");
+                    let mut mock = PeerMock::new(precompute_key.clone(), nonce.clone(), socket);
+                    if let Some(chunk_size) = chunk_size {
+                        mock.chunk_size(chunk_size);
+                    }
+                    mock.send_message(message.clone());
+                }
+                Err(e) if e.kind() == std::io::ErrorKind::WouldBlock => (),
+                Err(e) => panic!("error accepting a connection: {}", e),
+            }
+            match stop_rx.try_recv() {
+                Ok(()) => break,
+                Err(TryRecvError::Empty) => (),
+                Err(TryRecvError::Disconnected) => panic!("stop channel disconnected"),
+            }
+            std::thread::sleep(Duration::from_millis(10));
+        }
+    });
+
+    let rt = Builder::new_multi_thread()
+        .enable_io()
+        .build()
+        .expect("Cannot build tokio runtime");
+
+    rt.block_on(async move {
+        use tokio::net::TcpStream;
+
+        debug!(receiver_log, "Tokio started");
+
+        let addr = addr_recv.recv().unwrap();
+        let log = receiver_log.new(o!("receiver" => addr));
+        debug!(log, "Received socket address");
+
+
+        b.iter(move || {
+            let log = log.clone();
+            debug!(log, "Iteration started");
+            let (done_tx, done_rx) = channel();
+            let precompute_key = crypto_local.precompute_key.clone();
+            let nonce = crypto_local.nonce_pair.remote.clone();
+            debug!(log, "Spawning tokio...");
+            tokio::spawn(async move {
+                debug!(log, "Tokio spawned...");
+                let mut socket = TcpStream::connect(addr).await.unwrap();
+                let reader = MessageReaderBase {
+                    stream: socket.split().0,
+                };
+                let mut reader = EncryptedMessageReaderBase::new(
+                    reader,
+                    precompute_key,
+                    nonce,
+                    new_log()
+                );
+                reader.read_message::<PeerMessageResponse>().await.unwrap();
+                done_tx.send(()).unwrap();
+            });
+            done_rx.recv().unwrap();
+        });
+        stop_tx.send(()).unwrap();
+    });
+
+    sender.join().unwrap();
+}
+
+
+
+/*
```

Comment:
@akoptelov 
what about this commented code?
do we need it or could be removed?

Reply:
Yeah, some leftovers from copy-pasting test code... Removed.